include/deal.II/lac/tensor_product_matrix.h

```diff
@@ -430,7 +261,7 @@ class TensorProductMatrixSymmetricSum<dim, VectorizedArray<Number>, n_rows_1d>
 
 namespace internal
 {
-  namespace TensorProductMatrix
+  namespace TensorProductMatrixSymmetricSum
```

Comment:
I have moved all the logic into the `internal::TensorProductMatrixSymmetricSum` namespace. This will allow me to reuse the algorithms in a class that is designed to store a minimal set of matrices.